### PR TITLE
deployment manager and prototype bug fix

### PR DESCRIPTION
**Pull request by kunmingg**
1. Add kubeflow-core to config, and fix typos.
2. Make bootstrapper image use master branch as registry, till we reelase 0.2.
3. Make bootstrapper use params from config when create components. (Some params are not optional.)

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/872)

### CODE REVIEW

File: docs/gke/configs/env-kubeflow.sh [hunk 1]
```diff
@@ -11,14 +11,14 @@ export PROJECT=kubeflow
 
 # The name of the ip address as defined in cluster.jinja
 # We will reserve a GCP IP address to use for ingress to the cluster.
-export IP_NAME=static-ip
+# export IP_NAME=static-ip
```
Comment (jlewi) [inline docs/gke/configs/env-kubeflow.sh]: Just delete the commented out values.

**Comment by jlewi** [comment]
The PR description could be a little more descriptive.

File: bootstrap/cmd/bootstrap/app/server.go [hunk 2]
```diff
@@ -482,7 +493,7 @@ func Run(opt *options.ServerOption) error {
 		// if use k8s client-go API, would be quite verbose if we create all resources one by one.
 		// TODO: use API to create ks Components
 		log.Infof("Apply kubeflow Components...")
-		rawCmd := "ks show default | kubectl apply -f -"
+		rawCmd := "ks show default | kubectl apply --validate=false -f -"
```
Comment (jlewi) [inline bootstrap/cmd/bootstrap/app/server.go]: Why are you setting validate to false? This seems dangerous. Where are we supplying invalid config?

File: docs/gke/configs/env-kubeflow.sh [hunk 3]
```diff
@@ -9,25 +9,9 @@
...
-# We will reserve a GCP IP address to use for ingress to the cluster.
-export IP_NAME=static-ip
-
-# Set the namespace to the name of the namespace to deploy Kubeflow in.
-export NAMESPACE=kubeflow
-
 # Set config file to the YAML file defining your deployment manager configs.
 export CONFIG_FILE=cluster-${PROJECT}.yaml
-export PROJECT_CONFIG_FILE=project-${PROJECT}.yaml
+# export PROJECT_CONFIG_FILE=project-${PROJECT}.yaml
```
Comment (jlewi) [inline docs/gke/configs/env-kubeflow.sh]: Delete the commented out code.

File: bootstrap/cmd/bootstrap/app/server.go (hunk 2 above)
Comment (kunmingg) [reply bootstrap/cmd/bootstrap/app/server.go]: Without validate=false, kubectl was complaining about not able to find CRD definitions. Skipping validate is a shortcut, let me see if there's other ways to apply.

Comment (jlewi) [reply bootstrap/cmd/bootstrap/app/server.go]: Setting validate to false is highly undesirable. Not sure why that would be necessary 
Also why can't we just do

ks apply? Why do we need to pipe it to kubectl apply?

Comment (kunmingg) [reply bootstrap/cmd/bootstrap/app/server.go]: Found the cause of validate bug:
https://github.com/kubeflow/kubeflow/pull/889

ksonnet doesn't support apply from pod but kubectl does; we can remove pipe to kubectl once ksonnet add this feature.

**Comment by kunmingg** [comment]
@ankushagarwal rebased, thanks.

**Pull request merged**